docs/examples/userguide/language_basics/casting_python.pxd

```diff
@@ -0,0 +1,2 @@
+cdef extern from *:
+    ctypedef Py_ssize_t Py_intptr_t
```

Comment:
The ugly way to make this work in pure python is to use [numpy initialization](https://github.com/cython/cython/blob/efdf996dced075682d86b7e53a12b56b5ebfea38/Cython/Includes/numpy/__init__.pxd#L28).
i.e. `from numpy cimport Py_intptr_t`

But, since this tutorial does not focus on the `extern` syntax, it is not clear what it is and how it works, so we can move this type definition somewhere in `Cython/Includes/cpython`

What do you think @scoder?

Reply:
Add it to the includes, definitely.
We could also add it to the known C types.

Reply:
In this case, the pyx version will also need to use the new `cimport`

Reply:
@scoder I would generally suggest covering all of `Python.h` with type declarations. Not necessarily in this PR, but it will be more pleasant if there are already `ctypedef `s for the cpython.

Reply:
> @scoder I would generally suggest covering C API path of `Python.h` with type declarations. Not necessarily in this PR, but it will be more pleasant if there are already `ctypedef `s for the cpython.

I fail to understand what you are suggesting here.

Reply:
> I fail to understand what you are suggesting here.

I suggested to create [these](https://github.com/cython/cython/tree/master/Cython/Includes/cpython) includes, but it seems that many of them are already listed in those pxd-s.